docs: update Agent Server changelog through 0.10.0 stable (#4406)
langchain-ai/docs: src/langsmith/agent-server-changelog.mdx

</Update>

<Update label="2026-06-05" tags={["agent-server"]}>
## v0.10.0rc3

### Fixes
- Fixed protocol v2 event streaming against JS sidecar (remote) graphs, which were incorrectly served through the legacy reconstruction path. Remote graphs now use LangGraphJS's native v3 stream for v2 event-streaming runs, resolving tool calls not rendering, headless interrupts never executing or resuming, and `400: tool_use ids must be unique` errors on the final message after a resume.
</Update>

<Update label="2026-06-02" tags={["agent-server"]}>
## v0.10.0rc2

### Fixes
- Fixed protocol v2 runs on JS graphs failing silently. The sidecar rejected `streamEvents` with a 400 due to strict stream-mode validation, the error was swallowed, and runs falsely reported success with 0 nodes executed. Relaxed stream-mode validation at the HTTP boundary and now raise a clear error on non-2xx sidecar responses instead of masking the failure.
</Update>

<Update label="2026-06-01" tags={["agent-server"]}>
## v0.10.0rc1

### General Notes

> - v0.10.0rc1 includes breaking changes for security and correctness. See Security section below for more details.

Reviewer: ```suggestion
- v0.10.0rc1 includes breaking changes for security and correctness. Refer to the [Security section](#security) for more details.
```

### Security
- **Potentially breaking** Loopback webhook targets are now denied by default to fix an authentication-bypass primitive ([GHSA-2c9q-c2q9-qgqv](https://github.com/langchain-ai/helm/security/advisories/GHSA-2c9q-c2q9-qgqv)). The `webhooks.url.disable_loopback` policy now defaults to `true`, blocking relative-URL webhooks (which dispatch through the in-process ASGI transport and bypass auth), as well as localhost / 127.x / ::1 / host.docker.internal absolute URLs and any hostname that DNS-resolves into the loopback range (mitigating DNS rebinding). Deployments that legitimately need loopback webhooks (e.g. `langgraph dev` with a localhost webhook receiver, or production setups that dispatch to a custom FastAPI route mounted on the same server) can opt back in by setting `webhooks.url.disable_loopback: false` in `langgraph.json` (or the equivalent `LANGGRAPH_WEBHOOKS` JSON env var). Only do this when you control the routes that loopback webhooks reach, as those routes are dispatched without authentication.

> - **Potentially breaking** `POST /runs` and `POST /threads/{thread_id}/runs` now authorize the attached assistant via the`assistants.read` auth event (matching cron creation and direct GET) instead of the previously-used `assistants.search` event with an incomplete payload ([GHSA-jfj5-wrj9-63x4](https://github.com/langchain-ai/helm/security/advisories/GHSA-jfj5-wrj9-63x4)). Deployments that registered only `@auth.on.assistants.read` (and no `.search` handler) were vulnerable to a cross-user authorization bypass; their existing read handler will now be invoked on the run-creation path. As a defense-in-depth follow-up, client-supplied run/cron metadata is no longer forwarded into the `assistants.read` auth event payload from `Runs.put` or `Crons.put`, and inmem/postgres runtimes now agree on the value shape. **Breaking change for deployments with custom auth handlers**: (1) any `@auth.on.assistants.search` handler that was previously invoked during run creation is no longer called there — ensure you have an equivalent`@auth.on.assistants.read` handler returning the same owner-style filter; (2) `value["metadata"]` on the `assistants.read` event invoked from run/cron creation is no longer populated, so handlers that inspected or mutated it must move that logic into `@auth.on.runs.create_run` / `@auth.on.crons.create` and rely on returning a filter for ownership enforcement.

Reviewer: ```suggestion
- **Potentially breaking** `POST /runs` and `POST /threads/{thread_id}/runs` now authorize the attached assistant via the`assistants.read` auth event (matching cron creation and direct GET) instead of the previously used `assistants.search` event with an incomplete payload ([GHSA-jfj5-wrj9-63x4](https://github.com/langchain-ai/helm/security/advisories/GHSA-jfj5-wrj9-63x4)). Deployments that registered only `@auth.on.assistants.read` (and no `.search` handler) were vulnerable to a cross-user authorization bypass; their existing read handler will now be invoked on the run-creation path. As a defense-in-depth follow-up, client-supplied run/cron metadata is no longer forwarded into the `assistants.read` auth event payload from `Runs.put` or `Crons.put`, and inmem/postgres runtimes now agree on the value shape. **Breaking change for deployments with custom auth handlers**: (1) any `@auth.on.assistants.search` handler that was previously invoked during run creation is no longer called there. Ensure you have an equivalent`@auth.on.assistants.read` handler returning the same owner-style filter; (2) `value["metadata"]` on the `assistants.read` event invoked from run/cron creation is no longer populated, so handlers that inspected or mutated it must move that logic into `@auth.on.runs.create_run` / `@auth.on.crons.create` and rely on returning a filter for ownership enforcement.
```

Reviewer: Should the "**Breaking change for deployments with custom auth handlers**:" be its own bullet or sub-bullet?

Author: It should all be in that one bullet. I can probably unbold that section given the whole area is marked potentially breaking.
